barrels/Semicircles/Semicircles.mc

```diff
@@ -0,0 +1,257 @@
+//
+// Copyright 2017 by Garmin Ltd. or its subsidiaries.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+using Toybox.Position;
+using Toybox.Lang;
+using Toybox.Math;
+using Toybox.System;
+using Toybox.Test;
+
+module Semicircles {
+
+    //! Execption for unhandled coordinate formats
+    class BadCoordinateFormatException extends Lang.Exception {
+        private var _msg;
+        public function initialize(msg) {
+            Exception.initialize();
+            _msg = msg;
+        }
+
+        public function getErrorMessage() {
+            return _msg;
+        }
+    }
+
+
+    //! Class to represent a coordinate that does not use Position module
+    //! This class uses 32 bit semicircles, where PI radians are equal
+    //! to 0x80000000 32 bit semicircles. Semicircles handle wrap-around via
+    //! integer overflow, and all angle math can be done using integer math
+    //! speeding up computation. This class also includes a fast distance
+    //! between two point computation.
+    class Coordinate {
+        //! Constant for semicircles per meter at the equator
+        const SC_PER_M = 107.26d;
```

Comment:
Minor: but this value appears to be slightly off.  I think it should be 107.173188, but based on what I remember, this tended to just be rounded to 107 to avoid expensive floating point math.